### ISSUE TITLE
(Win2012r2) Chef-client version to install seems to be ignored

### ISSUE THREAD

**Issue by vinyar** [test-kitchen/test-kitchen#882]
__Bug:__
It doesn't seem to matter what version is specified in the .kitchen.yml file. That version is reported as having been detected.

__Example:__
.kitchen.yml
```yml
---
driver:
  name: vagrant
provisioner:
  require_chef_omnibus: 57.24.0
platforms:
  - name: win2012_chef11_14_2
suites:
  - name: default
    run_list:
      - recipe[ces_chefworkstation::default]
    attributes:
```
output:
```ruby
-----> Converging <default-win2012-chef11-14-2>...
       Preparing files for transfer
       Preparing dna.json
       Resolving cookbook dependencies with Berkshelf 4.0.1...
       Removing non-cookbook files before transfer
       Preparing solo.rb
-----> Chef Omnibus installation detected (57.24.0)
       Transferring files to <default-win2012-chef11-14-2>
....
       * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
       Starting Chef Client, version 11.14.2
       Compiling Cookbooks...
```

__Dev env info:__
$ chef -v
Chef Development Kit Version: 0.10.0
chef-client version: 12.5.1
berks version: 4.0.1
kitchen version: 1.4.2
$ vagrant plugin list
kitchen-nodes (0.5.0)
vagrant-azure (1.3.0)
vagrant-berkshelf (4.1.0)
vagrant-cachier (1.2.1)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.4, system)
vagrant-winrm (0.7.0)

Notes:
*  :latest resolves to correct latest version available on chef.io, however the error remains the same.
*  I poked around the code, and didnt see anything obvious. chef-solo -v returns correct version presently on the box `11.14.2` (https://github.com/chef/vagrant-omnibus/blob/00c19b5ba0c76c2a1684580d4c0c9db4703f0fd7/lib/vagrant-omnibus/action/install_chef.rb#L139)
* complete guess, but perhaps the error is in this variable: https://github.com/chef/vagrant-omnibus/blob/00c19b5ba0c76c2a1684580d4c0c9db4703f0fd7/lib/vagrant-omnibus/action/install_chef.rb#L45

reference: https://github.com/chef/vagrant-omnibus/issues/125

**Comment by icberg7** [comment]
This appears to be a duplicate of #847.

**Comment by mwrock** [comment]
yeah this definitely seems broken. However as of current kitchen master and future releases, the bug resides in the mixlib/install gem and a fix is in https://github.com/chef/mixlib-install/pull/21

**Comment by cheeseplus** [comment]
Blocked on https://github.com/chef/mixlib-install/pull/21

**Comment by cheeseplus** [comment]
Closing as it's a dupe of https://github.com/test-kitchen/test-kitchen/issues/882

**Comment by icberg7** [comment]
@cheeseplus This is #882. Did you mean to close this a duplicate of #847?

**Comment by vinyar** [comment]
@cheeseplus I just tested this with latest chefdk - and I'm getting same behavior. 
```
$ chef -v
Chef Development Kit Version: 0.11.2
chef-client version: 12.7.2
berks version: 4.2.0
kitchen version: 1.5.0
```

**Comment by mwrock** [comment]
I just tried this on:
chef-dk: 0.11.2
kitchen: 1.6.0
mixlib-install: 0.7.1

And a provisoner config:
```
provisioner:
  name: chef_zero
  require_chef_omnibus: 12.5.1
```

The provisioner successful downloaded and installed chef v12.5.1.

**Comment by vinyar** [comment]
@mwrock can you try setting the installer to lets say 12.1, run a converge, then set it to 12.5, or even something non existent as in the example?

There's gotta be something different between our setups.
```
$ kitchen converge
-----> Starting Kitchen (v1.5.0)
-----> Converging <default-win2012-chef11-14-2>...
       Preparing files for transfer
       Preparing dna.json
       Preparing cookbooks from project directory
       Removing non-cookbook files before transfer
       Preparing validation.pem
       Preparing client.rb
-----> Chef Omnibus installation detected (57.24.0)

       Transferring files to <default-win2012-chef11-14-2>
       Starting Chef Client, version 11.14.2
...
```

**Comment by mwrock** [comment]
I get this:
```
-----> Installing Chef Omnibus (57.24.0)

       Downloading package from

       Skipping md5 verification

       Installing Chef Omnibus package C:\Users\vagrant\AppData\Local\Temp\chef-57.24.0.msi

$$$$$$ Exception calling "DownloadString" with "1" argument(s): "The remote server
$$$$$$ returned an error: (404) Not Found."
```

Have you tried with TK 1.6.0? And what version of mixlib-install do you have?

**Comment by smurawski** [comment]
@vinyar, I get the same results as @mwrock with Test-Kitchen 1.6 and mixlib-install 0.7.1.  But I have seen that behavior on earlier versions of mixlib-install and test-kitchen before it used mixlib-install.

**Comment by vinyar** [comment]
with TK 1.6 I get the expected behavior
